src/util/promoHelpers.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+import { asArray, asBoolean, asDate, asObject, asOptional, asString, asValue } from 'cleaners'
+import { type EdgeDataStore } from 'edge-core-js'
+import { Platform } from 'react-native'
+import { getVersion } from 'react-native-device-info'
+import { getLocales } from 'react-native-localize'
+
+import { config } from '../theme/appConfig.js'
+
+const asPromotion = asObject({
+  messageId: asString,
+  message: asObject(
+    // keys are language identifiers (ie. 'enUS')
+    asObject({
+      title: asString,
+      body: asString,
+      imageUrl: asString
+    })
+  ),
+  locations: asObject(
+    // Keys are 2-character country codes (ie. 'US')
+    asBoolean
+  ),
+  appFlags: asObject(
+    // Keys are unique account attributes (ie. 'wyreLinkedBank')
+    asBoolean
+  ),
+  appId: asOptional(asString),
+  forPlatform: asValue('ios', 'android'),
+  maxVersion: asString,
+  minVersion: asString,
+  startDate: asDate,
+  endDate: asDate
+})
+
+export type Promotion = $Call<typeof asPromotion>
+
+type Profile = {
+  appId: string | void,
+  appVersion: string,
+  language: string,
+  location: string,
+  platform: 'ios' | 'android'
+}
+
+export type AppFlags = {
+  wyreLinkedBank?: boolean
+}
+
+export const getPromotions = async (): Promise<Promotion[]> => {
```

Comment:
This can be a hook 'usePromotions(profile: Profile)' which returns the filtered tested promoCards instead of a getter.

That way you abstract the entire logic out of the component and into the hook.